hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INodeFile.java

```diff
@@ -458,7 +452,16 @@ public FileWithSnapshotFeature addSnapshotFeature(FileDiffList diffs) {
     this.addFeature(sf);
     return sf;
   }
-  
+
+  /** Used by FSImage. */
+  public INodeFile loadSnapshotFeature(FileDiffList diffs) {
+    final FileWithSnapshotFeature sf = addSnapshotFeature(diffs);
```

Comment:
Since this was not persisted before, we are just making sure while loading back by having the below check. Is my understanding correct?

Reply:
That's is correct.